src/gt4py/backend/numpy_backend.py

```diff
@@ -242,12 +242,11 @@ def visit_TernaryOpExpr(self, node: gt_ir.TernaryOpExpr) -> str:
         then_fmt = "({})" if isinstance(node.then_expr, gt_ir.CompositeExpr) else "{}"
         else_fmt = "({})" if isinstance(node.else_expr, gt_ir.CompositeExpr) else "{}"
 
-        source = "vectorized_ternary_op(condition={condition}, then_expr={then_expr}, else_expr={else_expr}, dtype={np}.{dtype})".format(
+        source = "{np}.where({condition}, {then_expr}, {else_expr})".format(
```

Comment:
This is definitively cleaner than the current implementation, but is it always guaranteed to be correct? I mean, `condition`, `then_expr` and `else_expr` could have different shapes or dtypes, right?

Reply:
The condition is based on the field and the else is always the original field. I could not think of a way to have different a shape on the then-expression? How would that look?

Reply:
Ok, it makes sense. We had a slightly different parallel model _intuition_ for conditionals in the past that does not fit anymore. Since tests pass, I agree in accepting the PR as it is and change whatever it is needed when we complete the verification of the parallel model.